pkg/validation/strfmt/default.go

```diff
@@ -53,7 +53,8 @@ const (
 	//   - dashes are permitted, but not at the start or the end of a segment
 	//   - long top-level domain names (e.g. example.london) are permitted
 	//   - symbol unicode points are permitted (e.g. emoji) (not for top-level domain)
-	HostnamePattern = `^([a-zA-Z0-9\p{S}\p{L}]((-?[a-zA-Z0-9\p{S}\p{L}]{0,62})?)|([a-zA-Z0-9\p{S}\p{L}](([a-zA-Z0-9-\p{S}\p{L}]{0,61}[a-zA-Z0-9\p{S}\p{L}])?)(\.)){1,}([a-zA-Z\p{L}]){2,63})$`
+	//   - last segment can contain hypens and digits
+	HostnamePattern = `^([a-zA-Z0-9\p{S}\p{L}]((-?[a-zA-Z0-9\p{S}\p{L}]{0,62})?)|([a-zA-Z0-9\p{S}\p{L}](([a-zA-Z0-9-\p{S}\p{L}]{0,61}[a-zA-Z0-9\p{S}\p{L}])?)(\.)){1,}(-?[a-zA-Z0-9\p{S}\p{L}]){2,63})$`
```

Comment:
where is this used and what are the implications of loosening this validation for data that is served by multiple servers running different versions of this component?

Reply:
In our Openshift CI, we have some tests where we specify hostnames with hypens and digits. We noticed this failure recently when one of our operators was degraded due to this check failing. As for the implications,as you said, this loosens restrictions , so the hostname validation should be more open now. that being said, this does conform to the specifications in RFC 1034, so it should still be within bounds of allowing a valid hostname to pass through and still restricting the invalid ones.

Reply:
also just to clarify we hit this in some quirky CI setup where the networking is internal and there are hostnames which are not like regular fqdns.

Reply:
I haven't jumped through all the RFC references to verify relaxing this is correct, but even if it is, until we have something like https://github.com/kubernetes/kubernetes/issues/94060 in place for a release, I don't think we can safely relax this library's validation.

Reply:
would like @sttts' opinion as well

Reply:
+1 for ratcheting validation (in the previous version) as possible generic solution. Otherwise this leads to downgrade problems with existing, now less restrictive objects.

In places where we control the CRD in (all) older releases, we can also easiliy replace the format with this less restrictive regex in the CRD schema. So there is a workaround. Not sure which OpenShift CRD this is about, but chance is high it is controlled by RH.

Reply:
yes this is specific to RH : https://github.com/openshift/api/blob/master/config/v1/types_ingress.go#L127. I understand the argument. For now, we have changed the hostname for our tests. will close this out now and revisit later when we have the ratcheting validations. thanks!